src/main/java/gg/uhc/uhc/inventory/IconInventory.java

```diff
@@ -68,14 +96,7 @@ protected int add(IconStack icon) {
         // add item to list in the correct location
         final int index = indexToInsert(icon);
         icons.add(index, icon);
-
-        // make sure this is enough space now
-        ensureInventorySize();
-
-        // rerender all items past this one (inclusive) as they have shifted index right 1
-        for (int i = index; i < icons.size(); i++) {
-            inventory.setItem(i, icons.get(i));
-        }
+        reopenForCurrentViewers();
```

Comment:
Is the performance difference for 40+ recreations of the inventory per player on a /reload not going to be an issue? Could we at least debounce the add and weightupdate recreates so there isnt a lot of redundant recreations? 500ms should be good enough and delaying the view of them for half a second shouldnt cause issues